src/qt/clientmodel.cpp

```diff
@@ -226,31 +226,39 @@ static void BannedListChanged(ClientModel *clientmodel)
     assert(invoked);
 }
 
-static void BlockTipChanged(ClientModel *clientmodel, bool initialSync, int height, int64_t blockTime, double verificationProgress, bool fHeader)
+static ClientModel::NotificationStatus GetNotificationStatus (interfaces::Node& node, bool initial_sync)
 {
-    // lock free async UI updates in case we have a new block tip
-    // during initial sync, only update the UI if the last update
-    // was > 250ms (MODEL_UPDATE_DELAY) ago
-    int64_t now = 0;
-    if (initialSync)
-        now = GetTimeMillis();
-
-    int64_t& nLastUpdateNotification = fHeader ? nLastHeaderTipUpdateNotification : nLastBlockTipUpdateNotification;
+    if (node.getImporting()) return ClientModel::NotificationStatus::INIT_IMPORT;
+    if (node.getReindex()) return ClientModel::NotificationStatus::INIT_REINDEX;
+    if (initial_sync) return ClientModel::NotificationStatus::INIT_DOWNLOAD;
+    return ClientModel::NotificationStatus::POST_INIT;
+}
 
+static void BlockTipChanged(ClientModel *clientmodel, bool initialSync, int height, int64_t blockTime, double verificationProgress, bool fHeader)
+{
     if (fHeader) {
         // cache best headers time and height to reduce future cs_main locks
         clientmodel->cachedBestHeaderHeight = height;
         clientmodel->cachedBestHeaderTime = blockTime;
     }
 
-    // During initial sync, block notifications, and header notifications from reindexing are both throttled.
-    if (!initialSync || (fHeader && !clientmodel->node().getReindex()) || now - nLastUpdateNotification > MODEL_UPDATE_DELAY) {
-        //pass an async signal to the UI thread
+    const ClientModel::NotificationStatus status = GetNotificationStatus(clientmodel->node(), initialSync);
+    const bool throttle = (status == ClientModel::NotificationStatus::INIT_DOWNLOAD && !fHeader) ||
```

Comment:
I don't see how this could be right. It appears to be never calling numBlockChanged at all during POST_INIT state

Usage of throttle here is also potentially confusing. "throttle" bool should indicate when to slow down notifications, not when to send them

Reply:
Fixed.